webbpsf/webbpsf_core.py

```diff
@@ -221,10 +239,49 @@ def pupil_mask(self,name):
 
         self._pupil_mask = name
 
-
     def __str__(self):
         return "<{telescope}: {instrument_name}>".format(telescope=self.telescope, instrument_name=self.name)
 
+    @property
+    def detector(self):
+        """Detector selected for simulated PSF
+
+        Used in calculation of field-dependent aberrations. Must be
+        selected from detectors in the `detector_list` attribute.
+        """
+        return self._detector
+
+    @detector.setter
+    def detector(self, value):
+        if value.upper() not in self.detector_list:
+            raise ValueError("Invalid detector. Valid detector names are: {}".format(', '.join(self.detector_list)))
+        self._detector = value.upper()
+
+    @property
+    def detector_list(self):
+        """Detectors on which the simulated PSF could lie"""
+        return sorted(self._detectors.keys())
+
+    @property
+    def detector_position(self):
+        """The pixel position in (X, Y) on the detector"""
+        #return self._detectors[self._detector].field_position
+        return self._detector_position
+
+    @detector_position.setter
+    def detector_position(self, position):
+        for i in range(2):
```

Comment:
I find this somewhat opaque.. it alsopotentially hides unexpected behavior (e.g. assigning `inst.detector_position = (123, 456, 'cat', None)` would pass unremarked).

Here's my suggestion:

```
    @detector_position.setter
    def detector_position(self, position):
        try:
            x, y = map(int, position)
        except ValueError:
            raise ValueError("Detector pixel coordinates must be pairs of nonnegative numbers, not {}".format(position))
        if x < 0 or y < 0:
            raise ValueError("Detector pixel coordinates must be nonnegative integers")
        if x > self._detector_npixels - 1 or y > self._detector_npixels - 1:
            raise ValueError("The maximum allowed detector pixel coordinate value is {}".format(self._detector_npixels - 1))

        self._detectors[self._detector].field_position = (x, y)
```